v1/ast/term.go

```diff
@@ -25,7 +25,14 @@ import (
 	"github.com/open-policy-agent/opa/v1/util"
 )
 
-var errFindNotFound = errors.New("find: not found")
+var (
+	NullValue Value = Null{}
+
+	errFindNotFound = errors.New("find: not found")
+
+	unescapedLeftCurly = regexp.MustCompile(`([^\\])({)`)
```

Comment:
Why capture? Should this not be the same?
```suggestion
	unescapedLeftCurly = regexp.MustCompile(`[^\\]{`)
```
Hmm looking at this without the `()`, it does look kind of ambiguous. Is that why?

Reply:
That'll give `Expected "unescaped \\{ left curly brace." but got "unescaped\\{ left curly brace."`

i.e. space removed for reasons I didn't care enough about to find out about. I don't like the regex thing either, so I'll see if I can get something else to work.

Reply:
Turns out both solutions were flawed, as they would not escape a curly brace at position 0 in a string. Will cook something better up tonight.